server/api.go

```diff
@@ -84,13 +76,21 @@ func (h *handler) downloadDumpHandler(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusInternalServerError)
 		return
 	}
+	defer func() {
+		fErr := os.RemoveAll(filepath.Dir(fp))
+		if fErr != nil {
+			h.plugin.API.LogError("Unable to remove temp directory for the dump", "error", fErr.Error())
+		}
+	}()
 
-	b, err := h.plugin.fileBackend.ReadFile(fp)
+	f, err := os.Open(fp)
```

Comment:
So to clarify: you are creating the dump from S3 -> localstore. And then downloading the final file from local store to the user?

Reply:
Yes